### DISCUSSION TITLE
Help: Problems with "yarn gulp (release)", BaseModel-related

### DISCUSSION THREAD

**Discussion by leylines** [TerriaJS/terriajs#7292] [Q&A]
Hi there,

I want to build a new release, but i' get always:

`ERROR in node_modules/terriajs/lib/Models/SearchProviders/CatalogSearchProvider.ts:174:11
TS2345: Argument of type 'unknown[]' is not assignable to parameter of type 'BaseModel[]'.
  Type 'unknown' is not assignable to type 'BaseModel'.
    172 |       } else {
    173 |         await loadAndSearchCatalogRecursively(
  > 174 |           this.terria.modelValues,
        |           ^^^^^^^^^^^^^^^^^^^^^^^
    175 |           searchText.toLowerCase(),
    176 |           searchResults,
    177 |           resultMap

ERROR in node_modules/terriajs/lib/ReactViews/Map/Panels/SharePanel/BuildShareLink.ts:152:9
TS18046: 'model' is of type 'unknown'.
    150 |
    151 |   terria.modelValues.forEach((model) => {
  > 152 |     if (model.uniqueId === GlobeOrMap.featureHighlightID) return;
        |         ^^^^^
    153 |     const force = terria.workbench.contains(model);
    154 |     addModelStratum(terria, model, stratumId, force, initSource);
    155 |   });

ERROR in node_modules/terriajs/lib/ReactViews/Map/Panels/SharePanel/BuildShareLink.ts:153:45
TS2345: Argument of type 'unknown' is not assignable to parameter of type 'BaseModel'.
    151 |   terria.modelValues.forEach((model) => {
    152 |     if (model.uniqueId === GlobeOrMap.featureHighlightID) return;
  > 153 |     const force = terria.workbench.contains(model);
        |                                             ^^^^^
    154 |     addModelStratum(terria, model, stratumId, force, initSource);
    155 |   });
    156 |

ERROR in node_modules/terriajs/lib/ReactViews/Map/Panels/SharePanel/BuildShareLink.ts:154:29
TS2345: Argument of type 'unknown' is not assignable to parameter of type 'BaseModel'.
    152 |     if (model.uniqueId === GlobeOrMap.featureHighlightID) return;
    153 |     const force = terria.workbench.contains(model);
  > 154 |     addModelStratum(terria, model, stratumId, force, initSource);
        |                             ^^^^^
    155 |   });
    156 |
    157 |   // Go through knownContainerUniqueIds and make sure they exist in models`

I have tried undoing all my changes, but the errors remain.

The release works afaik.

Do you have any idea where these errors are coming from?

Thank you so much,

Jörg

**Comment by leylines** [comment]
With today's update everthing is okay again.

**Reply by leylines** [reply]
The topic can be deleted.

**Reply by ljowen** [reply]
Thanks for updating, closing this thread
